core/src/libtx/proof.rs

```diff
@@ -195,9 +195,7 @@ where
 		let mut msg = [0; 20];
 		msg[2] = switch as u8;
 		let id_bytes = id.to_bytes();
-		for i in 0..17 {
-			msg[i + 3] = id_bytes[i];
-		}
+		msg[3..(17 + 3)].clone_from_slice(&id_bytes[..17]);
```

Comment:
I'm guessing tests will pick up any regressions here hard to be 100% certain this is identical logic.

Reply:
The loop is simply copying the bytes from `id_bytes` from 0 to 17 to the msg structure. I actually remove the addition which I think is a bit confusing. But overall should be faster thanks to `memcpy`.